gtksourceview.c: In function 'set_source_buffer':
gtksourceview.c:1589:43: error: assignment to 'GtkSourceBuffer *' {aka 'struct _GtkSourceBuffer *'} from incompatible pointer type 'GtkTextBuffer *' {aka 'struct _GtkTextBuffer *'} [-Wincompatible-pointer-types]

diff -up ./gtksourceview/gtksourceview.c.orig ./gtksourceview/gtksourceview.c
--- ./gtksourceview/gtksourceview.c.orig	2024-04-25 19:07:06.911596640 -0400
+++ ./gtksourceview/gtksourceview.c	2024-04-25 19:07:31.644768498 -0400
@@ -1586,7 +1586,7 @@ set_source_buffer (GtkSourceView *view,
 	{
 		GtkSourceBufferInternal *buffer_internal;
 
-		view->priv->source_buffer = g_object_ref (buffer);
+		view->priv->source_buffer = g_object_ref (GTK_SOURCE_BUFFER (buffer));
 
 		g_signal_connect (buffer,
 				  "highlight-updated",
